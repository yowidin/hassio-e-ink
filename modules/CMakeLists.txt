add_subdirectory_ifdef(CONFIG_ZEPHYR_CPP zephyr-cpp)
add_subdirectory_ifdef(CONFIG_TL_EXPECTED expected)
