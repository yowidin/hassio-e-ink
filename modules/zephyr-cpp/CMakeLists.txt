zephyr_library_named(zephyr_cpp)

zephyr_include_directories(include/public)

zephyr_library_sources(
    src/error.cpp
    src/mutex.cpp

    src/drivers/gpio.cpp
    src/drivers/spi.cpp
)
