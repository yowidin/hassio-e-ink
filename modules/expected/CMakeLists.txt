if(WEST)
   execute_process(COMMAND ${WEST} list -f "{abspath}" expected
      OUTPUT_VARIABLE TL_CPP_EXPECTED_LIB_DIR
      RESULT_VARIABLE TL_CPP_EXPECTED_LIB_DIR_RESULT
      WORKING_DIRECTORY ${CMAKE_CURRENT_LIST_DIR})
   if(NOT "${TL_CPP_EXPECTED_LIB_DIR_RESULT}" STREQUAL "0")
      message(FATAL_ERROR "Expected library not found")
   endif()
   string(STRIP ${TL_CPP_EXPECTED_LIB_DIR} TL_CPP_EXPECTED_LIB_DIR)
   print(TL_CPP_EXPECTED_LIB_DIR)
else()
   message(FATAL_ERROR "Cannot locate the expected library")
endif()

zephyr_include_directories(${TL_CPP_EXPECTED_LIB_DIR}/include/)
